Sources/Core/ViewModel/FormFieldViewModel.swift

```diff
@@ -56,47 +62,82 @@ final class FormFieldViewModel<AS: SparkAttributedString>: ObservableObject {
         theme: Theme,
         feedbackState: FormFieldFeedbackState,
         title: AS?,
-        description: AS?,
+        helper: AS?,
         isTitleRequired: Bool = false,
         colorUseCase: FormFieldColorsUseCaseable = FormFieldColorsUseCase(),
         titleUseCase: FormFieldTitleUseCaseable = FormFieldTitleUseCase()
     ) {
         self.theme = theme
         self.feedbackState = feedbackState
-        self.description = description
+        self.helper = helper
         self.isTitleRequired = isTitleRequired
         self.colorUseCase = colorUseCase
         self.titleUseCase = titleUseCase
         self.colors = colorUseCase.execute(from: theme, feedback: feedbackState)
-        self.spacing = self.theme.layout.spacing.small
-        self.titleFont = self.theme.typography.body2
-        self.descriptionFont = self.theme.typography.caption
+        self.spacing = Self.spacing(from: theme)
+        self.titleFont = Self.titleFont(from: theme)
         self.titleColor = self.colors.title
-        self.descriptionColor = self.colors.description
+        self.helperFont = Self.helperFont(from: theme)
+        self.helperColor = self.colors.helper
+        self.infoFont = Self.infoFont(from: theme)
+        self.infoColor = self.colors.info
         self.setTitle(title)
     }
 
+    // MARK: - Setter
+
     func setTitle(_ title: AS?) {
         self.userDefinedTitle = title
         self.updateTitle()
     }
 
+    func setCounter(text: String?, limit: Int?) {
+        guard let limit else {
+            self.info = nil
+            return
+        }
+
+        self.info = "\(text?.count ?? 0)/\(limit)"
+    }
+
+    // MARK: - Private Update
+
     private func updateTitle() {
         self.title = self.titleUseCase.execute(title: self.userDefinedTitle, isTitleRequired: self.isTitleRequired, colors: self.colors, typography: self.theme.typography) as? AS
     }
 
     private func updateColors() {
-        self.colors = colorUseCase.execute(from: self.theme, feedback: self.feedbackState)
+        self.colors = self.colorUseCase.execute(from: self.theme, feedback: self.feedbackState)
         self.titleColor = self.colors.title
-        self.descriptionColor = self.colors.description
+        self.helperColor = self.colors.helper
+        self.infoColor = self.colors.info
     }
 
     private func updateFonts() {
-        self.titleFont = self.theme.typography.body2
-        self.descriptionFont = self.theme.typography.caption
+        self.titleFont = Self.titleFont(from: self.theme)
+        self.helperFont = Self.helperFont(from: self.theme)
+        self.infoFont = Self.infoFont(from: self.theme)
     }
 
     private func updateSpacing() {
-        self.spacing = self.theme.layout.spacing.small
+        self.spacing = Self.spacing(from: self.theme)
+    }
+
+    // MARK: - Static func
+
+    private static func titleFont(from theme: Theme) -> any TypographyFontToken {
+        theme.typography.body2
```

Comment:
don't forget returns

Reply:
I don't like this rule but Ok, I made the change :)